internal/xds/clients/xdsclient/clientimpl_watchers.go

```diff
@@ -62,15 +65,19 @@ func (c *XDSClient) WatchResource(typeURL, resourceName string, watcher Resource
 	rType, ok := c.config.ResourceTypes[typeURL]
 	if !ok {
 		logger.Warningf("ResourceType implementation for resource type url %v is not found", rType.TypeURL)
-		watcher.ResourceError(fmt.Errorf("ResourceType implementation for resource type url %v is not found", rType.TypeURL), func() {})
+		c.serializer.TrySchedule(func(context.Context) {
+			watcher.ResourceError(fmt.Errorf("ResourceType implementation for resource type url %v is not found", rType.TypeURL), func() {})
```

Comment:
Nit: this error could be slightly reframed as "no ResourceType implementation found for typeURL %q", thereby making it more concise and at the same time being friendly to the Go style guide (not beginning error strings with uppercase).